File: nexus/ui/src/api/folders/create.ts
import { fetchJson, readCsrfHeader } from "@nube/starter-client-ts";
import type { StarterClient } from "@nube/starter-client-ts";

import type { CreateFolderRequest, FolderSummary } from "@/api/types";

// `POST /api/v1/folders` — create a folder (400 if the parent is absent or in
// another tenant). Omit `parent_id` for a root folder.
export function createFolder(
  client: StarterClient,
  request: CreateFolderRequest,
): Promise<FolderSummary> {
  return fetchJson<FolderSummary>(client, `${client.apiPrefix}/folders`, {
    method: "POST",
    headers: { "content-type": "application/json", ...readCsrfHeader() },
    body: JSON.stringify(request),
  });
}

File: nexus/backend/crates/nexus-spi/src/dto/folder/create.rs
//! `POST /api/v1/folders` — create a folder.

use serde::{Deserialize, Serialize};
use utoipa::ToSchema;
use uuid::Uuid;

/// Create a folder. `parent_id` files it under another folder (in the same
/// tenant); omit it for a root folder.
#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, ToSchema)]
pub struct CreateFolderRequest {
    pub name: String,
    /// Parent folder id; omit/`null` for a root folder.
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub parent_id: Option<Uuid>,
}
